src/main/java/org/trigon/hardware/rev/spark/SparkMotor.java

```diff
@@ -1,134 +1,185 @@
+     * @param pidSlot             the PID slot to use
+     * @param arbFeedForward      the feed forward value
+     * @param arbFeedForwardUnits the units of the feed forward value
+     */
+    public void setReference(double value, SparkBase.ControlType controlType, int pidSlot, double arbFeedForward, SparkClosedLoopController.ArbFFUnits arbFeedForwardUnits) {
+        motorIO.setReference(value, controlType, pidSlot, arbFeedForward, arbFeedForwardUnits);
+    }
+
+    /**
+     * Sets the transmission period for a specific periodic frame on the motor controller.
+     * This method adjusts the rate at which the controller sends the frame, but the change is not saved permanently and will reset on powerup.
+     *
+     * @param periodMs the new transmission period in milliseconds
+     */
+    public void setPeriodicFramePeriod(int periodMs) {
+        motorIO.setPeriodicFrameTimeout(periodMs);
+    }
+
+    /**
+     * Stops the motor.
+     */
     public void stopMotor() {
         motorIO.stopMotor();
     }
 
-    public void setBrake(boolean brake) {
-        motorIO.setBrake(brake);
-    }
-
+    /**
+     * Sets the motor's inverted value.
+     *
+     * @param inverted should the motor be inverted
+     */
     public void setInverted(boolean inverted) {
         motorIO.setInverted(inverted);
     }
 
-    public void enableVoltageCompensation(double voltage) {
-        motorIO.enableVoltageCompensation(voltage);
-    }
-
-    public void setClosedLoopRampRate(double rampRate) {
-        motorIO.setClosedLoopRampRate(rampRate);
-    }
-
-    public void setSmartCurrentLimit(int limit) {
-        motorIO.setSmartCurrentLimit(limit);
-    }
-
-    public void setOpenLoopRampRate(double rampRate) {
-        motorIO.setOpenLoopRampRate(rampRate);
-    }
-
-    public void setPID(double p, double i, double d) {
-        motorIO.setPID(p, i, d);
-    }
-
-    public void setConversionsFactor(double conversionsFactor) {
-        motorIO.setConversionsFactor(conversionsFactor);
-    }
-
-    public void restoreFactoryDefaults() {
-        motorIO.restoreFactoryDefaults();
-    }
-
-    public void enablePIDWrapping(double minInput, double maxInput) {
-        motorIO.enablePIDWrapping(minInput, maxInput);
+    /**
+     * Applies the configuration to the motor.
+     *
+     * @param configuration the configuration to apply
+     * @param resetMode     whether to reset safe parameters before setting the configuration or not
+     * @param persistMode   whether to persist the parameters after setting the configuration or not
+     */
+    public void configure(SparkBaseConfig configuration, SparkBase.ResetMode resetMode, SparkBase.PersistMode persistMode) {
+        motorIO.configure(configuration, resetMode, persistMode);
     }
 
-    public void burnFlash() {
-        CompletableFuture.runAsync(() -> {
-            Timer.delay(5);
-            motorIO.burnFlash();
-        });
+    public void setSimulationGearbox(DCMotor gearbox) {
```

Comment:
the sim can use much more than just the gearbox - use the normal sim classes

Reply:
You're saying not to use the SparkSim or to use both?

Reply:
I'm talking about using the physics sim classes that we created to input info into the motor